source/PlayerInfo.cpp

```diff
@@ -1346,6 +1346,20 @@ void PlayerInfo::ReorderShip(int fromIndex, int toIndex)
 	shared_ptr<Ship> ship = ships[fromIndex];
 	ships.erase(ships.begin() + fromIndex);
 	ships.insert(ships.begin() + toIndex, ship);
+	// Check if the ship in the first position can be a flagship and is in the current system.
+	if(Preferences::Has("Automatically unpark flagship")
+			&& ships[0]->CanBeFlagship() && ships[0]->GetSystem() == this->GetSystem())
+	{
+		if(fromIndex == 0)
+		{
+			ships[toIndex]->SetIsParked(true);
+		}
+		else
+		{
+			ships[fromIndex]->SetIsParked(true);
+		}
```

Comment:
you don't need braces for single statements (but you can keep them if you prefer)
```suggestion
			ships[toIndex]->SetIsParked(true);
		else
			ships[fromIndex]->SetIsParked(true);
```